src/porcelain/install.ts

```diff
@@ -23,7 +23,10 @@ export default async function(pkgs: PackageSpecification[] | string[] | string,
   //TODO parallelize!
 
   pkgs = (await hydrate(pkgs)).pkgs
-  const { pending, installed } = await resolve(pkgs)
+  const resolved = await resolve(pkgs)
+  logger?.resolved?.(resolved)
```

Comment:
Only the porcelain has a resolution so the Logger should be extended in this file with that logging message rather than added to the plumbing Logger interface. Which also means the Logger that `run.ts` uses needs to refer to this one.